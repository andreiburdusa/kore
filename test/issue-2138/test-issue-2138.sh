#!/bin/sh
exec kore-exec \
    definition.kore \
    --pattern pgm.kore \
    --module TEST \
    --smt-timeout 40 \
    --smt z3 \
    --strategy all \
    --log-level \
    warning \
    --enable-log-timestamps \
    --solver-transcript \
    smt-transcript \
    --search searchFile.kore \
     \
    --searchType FINAL \
    "$@"
