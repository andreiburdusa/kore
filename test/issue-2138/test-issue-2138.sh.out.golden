/* Created: Kore.Internal.Conditional.makeAndPredicate */
/* Spa */
\and{SortGeneratedTopCell{}}(
    /* Created: Kore.Internal.Predicate.makeAndPredicate */
    /* Sfa */
    \and{SortGeneratedTopCell{}}(
        /* Sfa */
        \equals{SortBool{}, SortGeneratedTopCell{}}(
            /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{SortBool{}}("true"),
            /* Fl Fn D Sfa */
            Lbl'UndsEqlsEqls'Int'Unds'{}(
                /* Fl Fn D Sfa */
                Lbli'LParUndsCommUndsRParUnds'TEST'Unds'Int'Unds'Int'Unds'Int{}(
                    /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{SortInt{}}("1"),
                    /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{SortInt{}}("0")
                ),
                /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{SortInt{}}("0")
            )
        ),
        /* Sfa */
        \forall{SortGeneratedTopCell{}}(
            ConfigVar'Ques'I:SortInt{},
            /* Sfa */
            \equals{SortInt{}, SortGeneratedTopCell{}}(
                /* Fl Fn D Sfa */
                Lbli'LParUndsCommUndsRParUnds'TEST'Unds'Int'Unds'Int'Unds'Int{}(
                    /* Fl Fn D Sfa */ ConfigVar'Ques'I:SortInt{},
                    /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{SortInt{}}("0")
                ),
                /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{SortInt{}}("0")
            )
        )
    ),
    /* Spa */
    \equals{SortGeneratedTopCell{}, SortGeneratedTopCell{}}(
        /* Fl Fn D Sfa */ VarResult:SortGeneratedTopCell{},
        /* Fl Fn D Sfa Cl */
        Lbl'-LT-'generatedTop'-GT-'{}(
            /* Fl Fn D Sfa Cl */
            Lbl'-LT-'k'-GT-'{}(/* Fl Fn D Sfa Cl */ dotk{}()),
            /* Fl Fn D Sfa Cl */
            Lbl'-LT-'generatedCounter'-GT-'{}(
                /* Fl Fn D Sfa Cl */ /* builtin: */ \dv{SortInt{}}("0")
            )
        )
    )
)